lib/index.js

```diff
@@ -1556,10 +1549,15 @@ Stream.prototype.pick = function (properties) {
             push(err, x);
         }
         else if (_.isObject(x)) {
+            for (var k in x) {
+                if (hasOwn.call(x, k)) {
+                    properties.push(k);
```

Comment:
Can't you just do the `if(f(key, x[key]))` block here? What's the point of pushing to an array just to immediately iterate over it later?